### PR TITLE
fix: migrate from jest to node test and c8

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

This PR adds [borp](https://github.com/mcollina/borp) and removes jest

Closes https://github.com/fastify/fastify-passport/issues/1157

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
Please keep the `test` inside the root `test` folder.
It shouldn't move inside the `src`.

**Comment by mcollina** [comment]
You'll need to bump the minimum node.js version. You can do so by increasing the action version to v5 in https://github.com/fastify/fastify-passport/blob/53bb3b79ca78952c5c3d62bf36b252f5e9730f98/.github/workflows/ci.yml#L20.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ on:
 
 jobs:
   test:
-    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v3
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v5
```
Comment (climba03003) [inline .github/workflows/ci.yml:20]:
```suggestion
    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v5.0.0
```

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: tsconfig.json [hunk 2]
```diff
@@ -10,12 +13,14 @@
...
+    "forceConsistentCasingInFileNames": true,
+    "skipLibCheck": true,
+    "lib": ["ESNext"]
   },
-  "include": [
-    "src"
-  ]
-}
\ No newline at end of file
+  "include": ["src", "test"]
```
Comment (climba03003) [inline tsconfig.json:25]: Please remove the `test` here.
It is used to build the packages, and should not include `test`.

File: tsconfig.test.json [hunk 3]
```diff
@@ -1,10 +0,0 @@
-{
```
Comment (climba03003) [inline tsconfig.test.json:1]: Revert the removal of this file.

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -2,16 +2,16 @@
   "name": "@fastify/passport",
   "version": "2.5.0",
   "description": "Simple, unobtrusive authentication for Fastify.",
-  "main": "dist/index.js",
+  "main": "dist/src/index.js",
```
Comment (climba03003) [inline package.json:5]: Please revert the changes to built path.

**Review by gurgunday — COMMENTED** [review]
Lint is failing

**Review by gurgunday — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by gurgunday — APPROVED** [review]
Lgtm

**Comment by dancastillo** [comment]
hey @climba03003 I updated based on your comments could you ptal?

**Review by climba03003 — APPROVED** [review]

**Review by gurgunday — APPROVED** [review]

**Pull request merged**